source/Octopus.Tentacle.Tests/Integration/StartScriptCommandV2Builder.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Scripts;
+using Octopus.Tentacle.Util;
+
+namespace Octopus.Tentacle.Tests.Integration
+{
+    public class StartScriptCommandV2Builder
+    {
+        readonly List<ScriptFile> files = new List<ScriptFile>();
+        readonly List<string> arguments = new List<string>();
+        readonly Dictionary<ScriptType, string> additionalScripts = new Dictionary<ScriptType, string>();
+        StringBuilder scriptBody = new StringBuilder(string.Empty);
+        ScriptIsolationLevel isolation = ScriptIsolationLevel.FullIsolation;
+        TimeSpan scriptIsolationMutexTimeout = ScriptIsolationMutex.NoTimeout;
+        string scriptIsolationMutexName = nameof(RunningScript);
+        string taskId = Guid.NewGuid().ToString();
+        ScriptTicket scriptTicket = new ScriptTicket(Guid.NewGuid().ToString());
+        TimeSpan durationStartScriptCanWaitForScriptToFinish = TimeSpan.Zero;
+
+        public StartScriptCommandV2Builder WithScriptBody(string scriptBody)
+        {
+            this.scriptBody = new StringBuilder(scriptBody);
+            return this;
+        }
+
+        public StartScriptCommandV2Builder WithScriptBody(string windowsScript, string bashScript)
+        {
+            this.scriptBody = new StringBuilder(PlatformDetection.IsRunningOnWindows ? windowsScript : bashScript);
```

Comment:
Would this method be useful? It is possible for the thing building the `StartScriptCommandV2` to be on a different OS to the remote target.

Reply:
This is a copy of the original builder

Reply:
It is being used?

Reply:
oh sorry I see its in the integration tests project! ok makes sense.

Reply:
Perhaps a name like `WithScriptBodyForCurrentOS` would make it super clear.